### PR TITLE
add warning about order if both mean and scale set

### PR DESCRIPTION
**Pull request by sadolini**
Root cause analysis: If mean and scale are both set by user in command line user can think that order of arguments influence on order of applying. 

Solution: It doesn't influence, so add warning about it.

Ticket: 68036

Code:
* [x]  Comments **NA just add warning**
* [x]  Code style (PEP8) **NA just add warning**
* [x]  Transformation generates reshape-able IR **NA just add warning**
* [x]  Transformation preserves original framework node names **NA just add warning**

Validation:
* [x]  Unit tests **NA just add warning**
* [x]  Framework operation tests **NA just add warning**
* [x]  Transformation tests **NA just add warning**
* [x]  Model Optimizer IR Reader check **NA just add warning**

Documentation:
* [x]  Supported frameworks operations list **NA just add warning**
* [x]  Guide on how to convert the **public** model **NA just add warning**
* [x]  User guide update **NA just add warning**

### CODE REVIEW

**Review by evolosen — APPROVED** [review]

File: tools/mo/openvino/tools/mo/main.py [hunk 1]
```diff
@@ -258,6 +258,10 @@ def raise_ie_not_found():
                 raise Error('Incorrect saved model tag was provided. Specify --saved_model_tags with no spaces in it')
             argv.saved_model_tags = argv.saved_model_tags.split(',')
 
+    if argv.mean_values and (argv.scale or argv.scale_values):
```
Comment (nosovmik) [inline tools/mo/openvino/tools/mo/main.py:261]: Per my understanding expected behavior is that if user specified --mean_values, then --scale - it is correct order and warning is not needed. It should be shown only if user specified scale, then mean, e.g. `--scale <val> --mean_values <vals>`

Comment (sadolini) [reply tools/mo/openvino/tools/mo/main.py:261]: Check of arguments order makes logic much more complex. By default arguments parser don't guarantee saving arguments order. I don't think it is required, we have not "right" or "wrong" order, order of arguments just don't influence on order of applying and we warn about it if both arguments are in place. If somewhere logic of implementation will be changed, we just re-write message in warning.

Comment (nosovmik) [reply tools/mo/openvino/tools/mo/main.py:261]: + @slyalin Actually this is exactly what has been identified as requirement for 'MO' vs 'IE' preprocessing. Using IE API user is able to first scale input values, then apply means by simple writing `preprocess().scale(2).mean(3)`. And user may have wrong expectation passing `--scale 3 --mean_values 2` will do the same. But this expectation is incorrect and we should warn about this.
For regular case when user specifies mean, then scale, there is no warning needed, we shall have same behavior as before - `warning` means that user may do something wrong, but by specifying means, then scales they do everything correctly. @slyalin can you please comment if it is ok to always show warning in MO, if user specifies means and scales in one command line. For me it is very usual use case in computer vision, so unavoidable warning may be annoying

File: tools/mo/openvino/tools/mo/main.py [hunk 2]
```diff
@@ -258,6 +258,10 @@ def raise_ie_not_found():
...
+        log.error("Order of arguments mean and scale in command line does not affect on order of applying it to network. " +
+                  "Mean always applies first, then scale.", extra={'is_warning': True})
```
Comment (popovaan) [inline tools/mo/openvino/tools/mo/main.py:263]:
```suggestion
        log.error("Order of arguments mean and scale in command line does not affect the order of applying it to network. " +
                  "Mean is always applied first, then scale.", extra={'is_warning': True})
```

**Review by popovaan — APPROVED** [review]

**Review by nosovmik — APPROVED** [review]

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 3]
```diff
@@ -271,15 +271,18 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
                                    'value. When a list of inputs is overridden by the --input ' +
                                    'parameter, this scale ' +
                                    'is not applied for any input that does not match with ' +
-                                   'the original input of the model.')
+                                   'the original input of the model.' +
+                                   'If both --mean and --scale  are specified, ' +
+                                   'the mean is subtracted first and then scale is applied ' +
+                                   'regardless the order of options in command line.')
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:277]: please update this help in `Converting_Model.md` for these options. We have a copy of this text there.

Comment (sadolini) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:277]: done

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 4]
```diff
@@ -91,6 +91,9 @@ Framework-agnostic parameters:
                         is overridden by the --input parameter, this scale is
                         not applied for any input that does not match with the
                         original input of the model.
+						If both --mean and --scale are specified,
+                        the mean is subtracted first and then scale is applied
+                        regardless the order of options in command line.
```
Comment (avladimi) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model.md:96]:
```suggestion
                        regardless of the order of options in command line.
```

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 5]
```diff
@@ -140,6 +143,9 @@ Framework-agnostic parameters:
                         data[255,255,255],info[255,255,255]". The exact
                         meaning and order of channels depend on how the
                         original model was trained.
+						If both --mean_values and --scale_values are specified,
+                        the mean is subtracted first and then scale is applied
+                        regardless the order of options in command line.
```
Comment (avladimi) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model.md:148]:
```suggestion
                        regardless of the order of options in command line.
```

File: docs/MO_DG/prepare_model/convert_model/Converting_Model.md [hunk 6]
```diff
@@ -195,9 +201,9 @@ Usually neural network models are trained with the normalized input data. This m
  
 In the first case, the Model Optimizer generates the IR with required pre-processing layers and Inference Engine samples may be used to infer the model. 
  
-In the second case, information about mean/scale values should be provided to the Model Optimizer to embed it to the generated IR. Model Optimizer provides a number of command line parameters to specify them: `--scale`, `--scale_values`, `--mean_values`. 
+In the second case, information about mean/scale values should be provided to the Model Optimizer to embed it to the generated IR. Model Optimizer provides a number of command line parameters to specify them: `--mean`, `--scale`, `--scale_values`, `--mean_values`. 
 
-If both mean and scale values are specified, the mean is subtracted first and then scale is applied. Input values are *divided* by the scale value(s). 
+> **NOTE:** If both mean and scale values are specified, the mean is subtracted first and then scale is applied regardless the order of options in command line. Input values are *divided* by the scale value(s). If also `--reverse_input_channels` option is used, the mean will be applied first, then scale and after that reverse_input_channels.
```
Comment (avladimi) [inline docs/MO_DG/prepare_model/convert_model/Converting_Model.md:206]:
```suggestion
> **NOTE:** If both mean and scale values are specified, the mean is subtracted first and then scale is applied regardless of the order of options in command line. Input values are *divided* by the scale value(s). If also `--reverse_input_channels` option is used, the mean will be applied first, then scale and after that reverse_input_channels.
```

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 3 above)
Comment (avladimi) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:277]:
```suggestion
                                   'regardless of the order of options in command line.')
```

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 7]
```diff
@@ -320,7 +323,10 @@ def get_common_cli_parser(parser: argparse.ArgumentParser = None):
                                    'Can be defined for desired input of the model, for example: ' +
                                    '"--scale_values data[255,255,255],info[255,255,255]". ' +
                                    'The exact meaning and order ' +
-                                   'of channels depend on how the original model was trained.',
+                                   'of channels depend on how the original model was trained.' +
+                                   'If both --mean_values and --scale_values are specified, ' +
+                                   'the mean is subtracted first and then scale is applied ' +
+                                   'regardless the order of options in command line.',
```
Comment (avladimi) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:329]:
```suggestion
                                   'regardless of the order of options in command line.',
```

**Review by avladimi — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

**Pull request merged**